Frameworks/CoreGraphics/CGPath.mm

```diff
@@ -15,176 +15,126 @@
 //******************************************************************************
 
 #import "CGContextInternal.h"
-#import "CGPathInternal.h"
-#import "CGPathInternal.h"
-#include "LoggingNative.h"
 #import <CoreGraphics/CGBitmapContext.h>
-#import <CoreGraphics/CGColorSpace.h>
-#import <CoreGraphics/CGContext.h>
-#import <CoreGraphics/CGContext.h>
-#import <CoreGraphics/CGGeometry.h>
-#import <CoreGraphics/CGGeometry.h>
-#import <CoreGraphics/CGImage.h>
-#import <CoreGraphics/CoreGraphicsExport.h>
+#import <CoreGraphics/CoreGraphics.h>
 #import <Starboard.h>
 #import <StubReturn.h>
 #import <algorithm>
 #import <vector>
 
-static const wchar_t* TAG = L"CGPath";
+#import <CoreFoundation/CoreFoundation.h>
+#import <CFRuntime.h>
+#import "D2DWrapper.h"
+
+#include <COMIncludes.h>
+#import <wrl/client.h>
+#import <D2d1.h>
+#include <COMIncludes_End.h>
+#import <CFCPPBase.h>
 
-class BBox {
-public:
-    float x, y, x1, y1;
-    bool xset, yset, x1set, y1set;
+static const wchar_t* TAG = L"CGPath";
 
-    BBox() {
-        x = y = x1 = y1 = 0;
-        xset = yset = x1set = y1set = false;
+//
+static inline CGPoint __CreateCGPointWithTransform(CGFloat x, CGFloat y, const CGAffineTransform* transform) {
+    CGPoint pt{ x, y };
+    if (transform) {
+        pt = CGPointApplyAffineTransform(pt, *transform);
     }
 
-    void addPoint(float ptX, float ptY) {
-        if (ptX < x || !xset) {
-            x = ptX;
-            xset = true;
-        }
-        if (ptY < y || !yset) {
-            y = ptY;
-            yset = true;
-        }
+    return pt;
+}
 
-        if (ptX > x1 || !x1set) {
-            x1 = ptX;
-            x1set = true;
-        }
+using namespace std;
+using namespace Microsoft::WRL;
 
-        if (ptY > y1 || !y1set) {
-            y1 = ptY;
-            y1set = true;
-        }
+struct __CGPathImpl {
+    ComPtr<ID2D1PathGeometry> pathGeometry{ nullptr };
```

Comment:
nit: no need to initialize ComPtr with nullptr, the default constructor will do that.